crates/stages/src/error.rs

```diff
@@ -36,6 +36,9 @@ pub enum StageError {
     /// rely on external downloaders
     #[error("Invalid download response: {0}")]
     Download(String),
+    /// Invalid checkpoint passed to the stage
+    #[error("Invalid checkpoint: {0}")]
+    Checkpoint(u64),
```

Comment:
This is technically fatal, right? Since the pipeline will just load up the same checkpoint again

Reply:
Checkpoint is a new word, let's call it StageProgress too?